dotcom-rendering/src/web/components/FrontGrid.tsx

```diff
@@ -0,0 +1,554 @@
+		grid-template-rows: [content-start] repeat(2, auto);
+	}
+`;
+
+const wideLeftColumn = css`
+	${from.leftCol} {
+		grid-template-columns:
+			[viewport-start] minmax(0, 1fr)
+			[title-start]
+			repeat(3, 60px)
+			[title-end content-start]
+			repeat(10, 60px)
+			[hide-start]
+			60px
+			[hide-end content-end] minmax(0, 1fr) [viewport-end];
+	}
+`;
+
+const sectionContentStretchedRight = css`
+	${from.wide} {
+		grid-template-columns:
+			[viewport-start] minmax(0, 1fr)
+			[title-start]
+			repeat(3, 60px)
+			[title-end content-start]
+			repeat(12, 60px)
+			[hide-start]
+			60px
+			[hide-end content-end]
+			minmax(0, 1fr) [viewport-end];
+	}
+`;
+
+const containerStylesToggleable = css`
+	${from.leftCol} {
+		grid-template-rows: auto [content-start] repeat(2, auto);
+	}
+	${from.wide} {
+		grid-template-rows: [content-start] repeat(3, auto);
+	}
+`;
+
+const headlineContainerStyles = css`
+	grid-row-start: 1;
+	grid-column: title;
+	${from.leftCol} {
+		grid-row-end: span 2;
+	}
+
+	display: flex;
+	flex-direction: column;
+`;
+
+const headlineContainerBorders = css`
+	${from.leftCol} {
+		position: relative;
+		::after {
+			content: '';
+			display: block;
+			width: 1px;
+			top: 0;
+			height: 1.875rem;
+			right: -10px;
+			position: absolute;
+			background-color: ${neutral[86]};
+		}
+	}
+`;
+
+const paddings = css`
+	padding-top: ${space[2]}px;
+	padding-bottom: ${space[9]}px;
+`;
+
+const sectionShowHide = css`
+	grid-row-start: 1;
+	grid-column: hide;
+	justify-self: end;
+`;
+
+const sectionContent = css`
+	margin: 0;
+
+	.hidden > & {
+		display: none;
+	}
+
+	grid-column: content;
+
+	grid-row-start: content-start;
+	${from.leftCol} {
+		grid-row-end: span 2;
+	}
+	${from.wide} {
+		grid-row-end: -1;
+	}
+`;
+
+const sectionContentBorder = css`
+	position: relative;
+
+	${from.leftCol} {
+		::before {
+			content: '';
+			display: block;
+			width: 1px;
+			top: 0;
+			bottom: 0;
+			left: -10px;
+			position: absolute;
+			background-color: ${neutral[86]};
+		}
+	}
+`;
+
+const sectionContentPadded = css`
+	${from.tablet} {
+		margin-left: -10px;
+		margin-right: -10px;
+	}
+`;
+
+const sectionTreats = css`
+	display: none;
+
+	${from.leftCol} {
+		display: block;
+		align-self: end;
+		grid-row-start: -2;
+		grid-column: title;
+	}
+
+	.hidden > & {
+		display: none;
+	}
+`;
+
+/** element which contains border and inner background colour, if set */
+const decoration = css`
+	grid-row: 1 / -1;
+
+	grid-column: 1 / -1;
+	${from.tablet} {
+		grid-column: 2 / -2;
+	}
+
+	border-width: 1px;
+	border-color: ${neutral[86]};
+	border-style: none;
+`;
+
+/** only visible once content stops sticking to left and right edges */
+const sideBorders = css`
+	${from.tablet} {
+		margin: 0 -20px;
+		border-left-style: solid;
+		border-right-style: solid;
+	}
+`;
+
+const topBorder = css`
+	border-top-style: solid;
+`;
+
+/**
+ * # Front Grid
+ *
+ * Represents a horizontal slice of a page, also known as a container.
```

Comment:
i _think_ a container can contain multiple slices? it's been a while, but iirc container > slice[s] > facia-card[s]?

Reply:
Oh, that’s a good point, let’s use a different word to “slice”.